src/UnoFlipController.java

```diff
@@ -86,42 +105,62 @@ public void actionPerformed(ActionEvent e) {
                         null,
                         null);
 
-                Card.Colour colour;
-                if (result == JOptionPane.OK_OPTION) {
+                // Colour is selected and OK is pressed
+                if (result == JOptionPane.OK_OPTION)
+                {
+                    // Colour to set the new current colour of the game
+                    Card.Colour colour;
+                    // Parse the input given by the user from the drop-down to set the new colour
                     String option = (String) comboBox.getSelectedItem();
                     colour = Card.Colour.valueOf(option);
+                    // Sets the new current colour
                     this.model.setCurrentColour(colour);
+                    // Notify view to show status of the current colour
                     this.model.notifyViews();
                     break;
-                } else if (result == JOptionPane.CLOSED_OPTION){
-                    continue;
                 }
             }
 
-        } else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD)){
+        }
+        // Draw card command for the player to draw a card
+        else if(e.getActionCommand().equals(UnoFlipViewFrame.DRAW_CMD))
+        {
+            // Model playTurn is called with -1 to signify a self draw one
             this.model.playTurn(-1);
         }
-        else if(e.getActionCommand().equals(UnoFlipViewFrame.NEXT_CMD)) {
+        // Next turn command to go to the next player
+        else if(e.getActionCommand().equals(UnoFlipViewFrame.NEXT_CMD))
+        {
+            // Model nextTurn is called
             this.model.nextTurn();
-        } else if (e.getActionCommand().equals(UnoFlipViewFrame.CHALLENGE_CMD) ) {
-
-            boolean challenge;
+        }
+        // Challenge command for wild draw 2 cards, prompts user if they want to challenge
```

Comment:
Same thing here: Put inside conditional block, and describe what actually happened (context is a bit more important here because of the unique behaviour of the wild draw 2 card). Somethign like "a wild draw 2 card was played, so prompt the next player to challenge/not challenge"